frontend/src/locales/i18n.ts

```diff
@@ -0,0 +1,25 @@
+import i18n from 'i18next';
+import { initReactI18next } from 'react-i18next';
+import translation_zh from './zh.json';
+// import translation_en from './en.json';
+
+// 初始化 i18next
+i18n.use(initReactI18next).init({
+  resources: {
+    // en: {
+    //   translation: translation_en
+    // },
+    zh: {
+      translation: translation_zh
+    }
+  },
+  lng: localStorage.getItem('locale') || 'en',
+  interpolation: {
+    escapeValue: false
+  }
+});
+
+// 将全局翻译方法绑定到window对象上
+(window as any).$t = i18n.t
```

Comment:
> The issue using i18n.t is that page needs to be reloaded to perform translation. With React applications it is recommended to use WithTranslation function (Class components) and useTranslation hooks (Function components)

The original purpose of using a globally bound method is to separate translations from code as much as possible. I think using the useTranslation hooks is also a good approach, but for some static files, it may increase the difficulty of translation. With a large number of project files, using a global approach eliminates the need to consider where it is being used specifically, what do you think?

Reply:
I understand your reason and it does make sense. The issue of using i18n.t is that translations won't be updated when you change the locale (for this reason you reload the whole page when you change the locale). On the other hand, useTranslation hook and WithTranslation function update the translations because they are part of the rendering process (see https://react.i18next.com/guides/the-drawbacks-of-other-i18n-solutions).